core/iwasm/compilation/aot_compiler.c

```diff
@@ -192,11 +192,18 @@ aot_compile_func(AOTCompContext *comp_ctx, uint32 func_index)
     while (frame_ip < frame_ip_end) {
         opcode = *frame_ip++;
 
+#if WASM_ENABLE_DEBUG_AOT != 0 || WASM_ENABLE_DYNAMIC_PGO != 0
 #if WASM_ENABLE_DEBUG_AOT != 0
         location = dwarf_gen_location(
             comp_ctx, func_ctx,
             (frame_ip - 1) - comp_ctx->comp_data->wasm_module->buf_code);
         LLVMSetCurrentDebugLocation2(comp_ctx->builder, location);
+#else
```

Comment:
IIUC, we are talking about two scenarios
- in WAMRC. There is no need to enable `WASMA_ENABLE_DYNAMIC_PGO`. we should `add_definitions(-DWASM_ENABLE_DYNAMIC_PGO=0)` in *wamr-compiler/CMakeLists.txt*
- in IWASM. Because both *DEBUG_AOT* and *DYNAMIC_PGO* will add `DILocation` into IR, we should only let one set `LLVMSetCurrentDebugLocation2` according to the current `module_type`.

So, how about?

```suggestion
#if WASM_ENABLE_DEBUG_AOT != 0
        if (wasm_module->module_type == Wasm_Module_AoT) {
            location = dwarf_gen_location(
                comp_ctx, func_ctx,
                (frame_ip - 1) - comp_ctx->comp_data->wasm_module->buf_code);
            LLVMSetCurrentDebugLocation2(comp_ctx->builder, location);
        }
#endif

#if WASM_ENABLE_DYNAMIC_PGO != 0
        if (wasm_module->module_type == Wasm_Module_Bytecode) {
            location = LLVMDIBuilderCreateDebugLocation(
                comp_ctx->context, frame_ip - func_ctx->aot_func->code, 1,
                func_ctx->debug_func, NULL);
            LLVMSetCurrentDebugLocation2(comp_ctx->builder, location);
        }
#endif
```

Plus, `add_definitions(-DWASM_ENABLE_DYNAMIC_PGO=0)` in *wamr-compiler/CMakeLists.txt*

Reply:
may be add a conflict protection in *build-scripts/config_common.cmake*?